.github/workflows/cigh.yml

```diff
@@ -0,0 +1,124 @@
+# This workflow aims at building and testing FreeCAD
+# It uses a runner based on latest Ubuntu LTS version
+# It builds using CMake/CCache+Make/Clang as a toolchain
+#
+# It has been foreseen to be used with CMake/Ninja/Clang...
+# ...but this is disabled at the moment because Ninja...
+# ...only works based on file timestamps, which isn't...
+# ...compatible with Git not managing them
+
+name: FCBuildNTest
+
+on:
+  workflow_dispatch: # Triggered manually
+# To run manually, use :
+# curl -X POST -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/:OWNER/FreeCAD/actions/workflows/cigh.yml/dispatches -d '{"ref":":BRANCH_NAME"}' -u :USER_NAME
+# It will then require your Github PAT to trigger the launching event
+    branches-ignore: # Do not run on master and release branches
+      - master
+      - releases/**
+  pull_request:
+    branches-ignore: # Do not run on master and release branches
+      - master
+      - releases/**
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  BuildNTest:
+    runs-on: ubuntu-latest
```

Comment:
The idea of the Docker container is that is saves a lot of times because it already has all build dependencies installed, that you would have to install on the GH container.
Also the container is a fixed image that lib versions are stable and known.
The drawback is that you have to update it on a regular base.